portfolio/src/main/webapp/script.js

```diff
@@ -1,34 +1,299 @@
+  addTerminalPrompt();
+}
+
+/*
+ * Lists all the commands on the terminal
+ */
+function whoamiCommand() {
+  catCommand('whoami');
+}
+
+/*
+ * Renders a selected site from the list
+ * @param {string} commandArgument Argument used in the command, expects a filename
+ */
+function catCommand(commandArgument) {
+  var newSection = '';
+
+  switch (commandArgument) {
+    case 'education.yaml':
+      newSection = 'education';
+      break;
+    case 'work-experience.yaml':
+      newSection = 'work-experience';
+      break;
+    case 'projects.yaml':
+      newSection = 'projects';
+      break;
+    case 'skills.yaml':
+      newSection = 'skills';
+      break;
+    case 'virus.exe':
+      newSection = 'interests';
+      break;
+    case 'whoami':
+      newSection = 'whoami';
+      break;
+    default:
+      printTerminalError(
+        'cat: the file selected does not exist',
+        commandArgument
+      );
+      return;
+  }
+
+  document.getElementById(currentSection).style.display = 'none';
+  document.getElementById(newSection).style.display = 'block';
+
+  currentSection = newSection;
+
+  addTerminalPrompt();
+}
+
+/*
+ * Explains what a command does on the terminal
+ * @param {string} commandArgument Argument used in the command, expects a command
+ */
+function manCommand(commandArgument) {
+  switch (commandArgument) {
+    case 'man':
+      addTerminalLine('man : PROVIDES DETAILS OF A COMMAND - man [command]');
+      break;
+    case 'compgen':
+      addTerminalLine('compgen : LISTS ALL AVAILABLE COMMANDS');
+      break;
+    case 'ls':
+      addTerminalLine('ls : LISTS ALL FILES');
+      break;
+    case 'cat':
+      addTerminalLine('cat : SEE CONTENTS A FILE - cat [filename]');
+      break;
+    case 'whoami':
+      addTerminalLine('whoami : SHOWS INFORMATION ABOUT USER');
+      break;
+  }
+  addTerminalPrompt();
+}
+
+/*
+ * Adds a new line to the terminal
+ * @param {string} text Text to add to the line
+ * @return {object}
+ */
+function addTerminalLine(text) {
+  var line = document.createElement('div');
+  line.className = 'terminal-line';
+  innerSpan = document.createElement('span');
+  innerSpan.textContent = text;
+  line.appendChild(innerSpan);
+  document.getElementById('terminal').appendChild(line);
+  scrollTerminalToBottom();
+}
+
+/*
+ * Prints some text to the terminal
+ * @param {string} message Message to print
+ * @param {function} callback Callback
+ */
 function printTerminal(message) {
-  //Simulates someone wrote a line to the terminal
-  var typeSpeed = 50;
-  var firstMessage = new Typed('#t1', {
-    strings: ['hola'],
-    typeSpeed,
-    startDelay: 1000,
+  var typeSpeed = 21;
+  var delayAtEnd = '1000';
+
+  var line = addTerminalLine();
+
+  return new Promise(function (resolve) {
+    //Writes to the inner element from the component recently created
+    var firstMessage = new Typed(
+      document.getElementById('terminal').lastChild.lastChild,
```

Comment:
Both here and in other places: Having multiple lastChild references can be somewhat hard to follow. Maybe when you create the element you can add an id to it and select that instead?

Reply:
Resolved via chat.